File: app/src/main/java/com/itverse/futuris/data/daos/ElementsDao.java
package com.itverse.futuris.data.daos;

import androidx.lifecycle.LiveData;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Query;

import com.itverse.futuris.data.entities.Element;

import java.util.List;

public interface ElementsDao {
    @Insert(onConflict = OnConflictStrategy.IGNORE)
    void insertElement(Element element);
    @Delete
    void deleteElement(Element element);
    @Query("SELECT * FROM element_table")
    LiveData<List<Element>> getAllElements();
    @Query("SELECT * FROM element_table WHERE element_name = :elementName")
    LiveData<Element> getElementByName(String elementName);


}

File: app/src/main/java/com/itverse/futuris/data/daos/ElementsDao.kt
package com.itverse.futuris.data.daos

import androidx.lifecycle.LiveData
import androidx.room.*
import com.itverse.futuris.data.entities.Element
import com.itverse.futuris.data.entities.relations.ComposantWithGroupedElements
import kotlinx.coroutines.flow.Flow

@Dao
interface ElementsDao {
    @Insert(onConflict = OnConflictStrategy.IGNORE)
    suspend fun insert(element: Element)

    @Delete
    suspend fun delete(element: Element)

    @get:Query("SELECT * FROM element_table")
    val allElements: LiveData<List<Element>>

    @Query("UPDATE element_table SET value = :value WHERE id = :elementId")
    suspend fun updateQuantity(elementId: Long, value: Int): Int

    @Query("SELECT * FROM element_table WHERE name = :name")
    fun getElementByName(name: String): LiveData<Element>
}